### PR TITLE
Returning appropriate status code for missing content

### LINKED ISSUE #348: 500 GET to /clusters/{cluster_id}/report
Smart proxy returns 500 status code when you do GET to `/clusters/{cluster_id}/report`. Log excerpt:

```
{"level":"info","time":"2020-11-12T15:39:19Z","message":"Request received - URI: /api/insights-results-aggregator/v1/clusters/5d5892d3-1f74-4ccf-91af-548dfc9767aa/report?osd_eligible=false, Method: GET"}
{"level":"error","error":"Item with ID ccx_rules_ocp.external.rules.machineconfig_stuck_by_node_taints/NODE_HAS_TAINTS_APPLIED was not found in the storage","time":"2020-11-12T15:39:19Z","message":"unable to get content for rule with id ccx_rules_ocp.external.rules.machineconfig_stuck_by_node_taints.report and error key NODE_HAS_TAINTS_APPLIED"}
{"level":"error","error":"unable to find content for rules","time":"2020-11-12T15:39:19Z","message":"handleServerError()"}
```

### PR DESCRIPTION
**Pull request by joselsegura**
# Description

Return a more appropriate status code

Fixes #348 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Testing steps

CI

### CODE REVIEW

File: config.toml [hunk 1]
```diff
@@ -3,7 +3,7 @@ address = ":8081"
 api_prefix = "/api/v1/"
 api_spec_file = "openapi.json"
 debug = true
-auth = false
```
Comment (quarckster) [inline config.toml:6]: Why did you change this?

**Review by tisnik — APPROVED** [review]
Looks good, thank you.

File: config.toml (hunk 1 above)
Comment (joselsegura) [reply config.toml]: For testing, I forgot to restore it before committing. Fixed

**Review by quarckster — APPROVED** [review]
